monkey/infection_monkey/credential_store/credentials_store.py

```diff
@@ -0,0 +1,29 @@
+from typing import Mapping
+
+from .i_credentials_store import ICredentialsStore
+
+
+class CredentialsStore(ICredentialsStore):
+    def __init__(self, credentials: Mapping = None):
+        self.stored_credentials = credentials
+
+    def add_credentials(self, credentials_to_add: Mapping) -> None:
+        if self.stored_credentials is None:
+            self.stored_credentials = {}
+
+        for key, value in credentials_to_add.items():
+            if key not in self.stored_credentials:
+                self.stored_credentials[key] = []
+
+            if key != "exploit_ssh_keys":
+                self.stored_credentials[key] = list(
+                    sorted(set(self.stored_credentials[key]).union(credentials_to_add[key]))
+                )
+            else:
+                self.stored_credentials[key] += credentials_to_add[key]
+                self.stored_credentials[key] = [
+                    dict(s) for s in set(frozenset(d.items()) for d in self.stored_credentials[key])
+                ]
```

Comment:
I'm not sure about any of this logic. What is `key`? Maybe use more descriptive names. It's not clear to me whether arranging and storing the data this way will get us where we need to go.